content/articles/angular-heat-maps/index.md

```diff
@@ -0,0 +1,447 @@
+        'Laureen', 'marrie', 'Paula', 'Karim', 'Marion'],
+};
+yAxis: Object = {
+    labels: ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat'],
+};
+}
+```
+In the above example, the heat map title is set to “Sales Revenue per Employee (in 1000 US$)”.
+### Enabling legend
+Apply a legend for the heat map in the `legendSettings` item by setting the `visible` characteristic to true and injecting the `legend` module by use of the `HeatMap.Inject(Legend)` technique.
+
+```typescript
+import { Component, ViewEncapsulation } from '@angular/core';
+import { HeatMap } from '@syncfusion/ej2-heatmap';
+
+@Component({
+selector: 'my-app',
+template:
+   `<ejs-heatmap id='container' style="display:block;"  [titleSettings]='titleSettings' [dataSource]='dataSource'
+   [xAxis]='xAxis' [yAxis]='yAxis' [legendSettings]='legendSettings'>
+    </ejs-heatmap>`,
+encapsulation: ViewEncapsulation.None
+})
+export class AppComponent{
+ titleSettings: Object = {
+    text: 'Sales Revenue per Employee (in 1000 US$)',
+    textStyle: {
+        size: '15px',
+        fontWeight: '500',
+        fontStyle: 'Normal'
+    }
+};
+    // Data for heatmap
+     dataSource: Object[] = [
+       [73, 39, 26, 39, 94, 0],
+       [93, 58, 53, 38, 26, 68],
+       [99, 28, 22, 4, 66, 90],
+       [14, 26, 97, 69, 69, 3],
+       [7, 46, 47, 47, 88, 6],
+       [41, 55, 73, 23, 3, 79],
+       [56, 69, 21, 86, 3, 33],
+       [45, 7, 53, 81, 95, 79],
+       [60, 77, 74, 68, 88, 51],
+       [25, 25, 10, 12, 78, 14],
+       [25, 56, 55, 58, 12, 82],
+       [74, 33, 88, 23, 86, 59]
+    ];
+    xAxis: Object = {
+    labels: ['Nancy', 'Jane', 'Janet', 'Margaret', 'Stephanie', 'Mitchel', 'Roberto',
+        'Laureen', 'Anne', 'Paula', 'Karim', 'Marion'],
+};
+yAxis: Object = {
+    labels: ['Mon', 'Tues', 'Wed', 'Thurs', 'Fri', 'Sat'],
+};
+public legendSettings: Object = {
+    visible:true,
+    position: 'Right',
+    showLabel:true,
+    height:'150px'
+};
+}
+```
+### Addition of the data label
+Include data labels on the heat map to improve its legibility. This can be accomplished by changing the `showLabel   ` property to` true` in the ` cellSettings ` object.
+
+
+```typescript
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+import { enableProdMode } from '@angular/core';
+import { AppModule } from './app.module';
+
+enableProdMode();
+platformBrowserDynamic().bootstrapModule(AppModule);
+
+
+```
+
+### Addition of custom palettes
+```typecript
+import { Component, ViewEncapsulation } from '@angular/core';
+import { HeatMap } from '@syncfusion/ej2-heatmap';
+
+@Component({
+selector: 'my-app',
+template:
+   `<ejs-heatmap id='container' style="display:block;" [paletteSettings]='paletteSettings'
+   [cellSettings]='cellSettings' [titleSettings]='titleSettings' [dataSource]='dataSource'
+   [xAxis]='xAxis' [yAxis]='yAxis' [legendSettings]='legendSettings'>
+    </ejs-heatmap>`,
+encapsulation: ViewEncapsulation.None
+})
```

Comment:
Hi, kindly explain this part well, like give readers with a detailed explanation on what the code does.